### PR TITLE
Add Moment Flatpak

### PR DESCRIPTION
**Pull request by Newbytee**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub.
- [X] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

Please invite @MRAAGH as a co-maintainer of the repo when it's merged.

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by Newbytee** [comment]
bot, build xyz.mx_moment.moment

**Comment by Newbytee** [comment]
bot, build xyz.mx_moment.moment

**Comment by Newbytee** [comment]
bot, build xyz.mx_moment.moment

File: xyz.mx_moment.moment.yaml [hunk 1]
```diff
@@ -0,0 +1,783 @@
...
+    sha256: db7aa48c0758ee45d7dbc1fde499912ec6fb77eab77a6e2808825d1f41d4e300
+
+- "python3-matrix-nio.json"
+
+- name: moment
+  buildsystem: qmake
+  post-install:
+  # This works around a bug in appstream-glib where it won't find the icon if
+  # /usr exists. For some reason.
+  - ln -s /app/share /app/usr
```
Comment (hfiguiere) [inline xyz.mx_moment.moment.yaml:775]: can't you instead patch the build system to no put stuff in `usr`?

Comment (Newbytee) [reply xyz.mx_moment.moment.yaml:775]: I'm not sure how to do it. I stole this workaround from Pure Maps: https://github.com/flathub/io.github.rinigus.PureMaps/blob/2d8797f2837774b11e7b3c2eab3946622ba8948e/io.github.rinigus.PureMaps.json#L160-L162

Comment (Newbytee) [reply xyz.mx_moment.moment.yaml:775]: Is there any chance we could let this be like this? It is a bit hacky, but I don't see any negative side-effects of it. I can try to look into it later, but we would like to formally announce Moment soon and we'd like to have a Flatpak easily available when we do. I tried looking into the qmake files for pyotherside (which is the guilty dependency) and I couldn't find any references to "usr".

Comment (hfiguiere) [reply xyz.mx_moment.moment.yaml:775]: it's not a blocker.

Comment (hfiguiere) [reply xyz.mx_moment.moment.yaml:775]: it's not a blocker

Comment (Newbytee) [reply xyz.mx_moment.moment.yaml:775]: Let's leave it as-is for now then.

File: xyz.mx_moment.moment.yaml [hunk 2]
```diff
@@ -0,0 +1,783 @@
...
+  cleanup:
+  - /bin
+  - /include
+  - /lib/pkgconfig
+  - /lib/*.la
+  sources:
+  - type: archive
+    url: https://mediaarea.net/download/source/libmediainfo/19.09/libmediainfo_19.09.tar.xz
+    sha256: ff06e1a449dfbe6f2c51f27ae1187d3e72386cb54476fbb189ffaacf845f478e
+- name: python3-cffi
```
Comment (barthalion) [inline xyz.mx_moment.moment.yaml:78]: Do these come from the pip generator? Could they be split to separate file?

File: xyz.mx_moment.moment.yaml [hunk 3]
```diff
@@ -0,0 +1,783 @@
+id: xyz.mx_moment.moment
+runtime: org.kde.Platform
+sdk: org.kde.Sdk
+runtime-version: '5.15-21.08'
+command: moment
+finish-args:
+- --share=ipc
+- --share=network
+- --socket=x11
```
Comment (barthalion) [inline xyz.mx_moment.moment.yaml:9]:
```suggestion
- --socket=fallback-x11
```

File: xyz.mx_moment.moment.yaml [hunk 4]
```diff
@@ -0,0 +1,783 @@
...
+- --socket=wayland
+- --socket=pulseaudio
+- --device=dri
+- --talk-name=org.freedesktop.Notifications
+rename-icon: moment
+rename-desktop-file: moment.desktop
+rename-appdata-file: moment.metainfo.xml
```
Comment (barthalion) [inline xyz.mx_moment.moment.yaml:16]: Ideally, these should happen upstream.

File: xyz.mx_moment.moment.yaml (hunk 2 above)
Comment (Newbytee) [reply xyz.mx_moment.moment.yaml:78]: Honestly I'm not sure, but I'd guess so. I've "inherited" most of this manifest from a previous inactive upstream. I could try splitting it out, sure, but I had issues when using flatpak-pip-generator to generate the entire list of dependencies.

File: xyz.mx_moment.moment.yaml (hunk 4 above)
Comment (Newbytee) [reply xyz.mx_moment.moment.yaml:16]: Are you suggesting that the files should use the ID format upstream too?

File: xyz.mx_moment.moment.yaml (hunk 2 above)
Comment (barthalion) [reply xyz.mx_moment.moment.yaml:78]: That's fair, but please look into it when you have spare time.

File: xyz.mx_moment.moment.yaml (hunk 4 above)
Comment (barthalion) [reply xyz.mx_moment.moment.yaml:16]: The original files should use the ID format. It's not a blocker, just mentioning that `rename` options are workarounds, not solutions.

**Pull request closed without merging**